### PR TITLE
fab now supports a size property

### LINKED ISSUE #290: [Enhancement] Change close button
**Is your enhancement request related to a problem? Please describe.**
Right now the modal has a big green close button, when its placed outside the modal. That close button should be a white 40x40 button as seen in the screenshot below.

Skærmbillede 2019-05-21 kl  08 26 48

### PR DESCRIPTION
**Pull request by mmmtrifork**
closes #290

### CODE REVIEW

File: src/kirby/components/floating-action-button/floating-action-button.component.ts [hunk 1]
```diff
@@ -9,4 +9,5 @@ export class FloatingActionButtonComponent {
   // TODO: showShadow should become an "elevation" enum in the future;
   @Input() showShadow?: boolean = true;
   @Input() disabled?: boolean = false;
+  @Input() size?: 'sm' | 'md' | 'lg' = 'sm';
```
Comment (unnamjolner) [inline src/kirby/components/floating-action-button/floating-action-button.component.ts]: We normally use the short versions: 's', 'm' and 'l'

File: src/kirby/components/floating-action-button/floating-action-button.component.tns.ts [hunk 2]
```diff
@@ -22,6 +22,7 @@ export class FloatingActionButtonComponent extends ContentView {
   // TODO: showShadow should become an "elevation" enum in the future;
   @Input() showShadow?: boolean = true;
   @Input() disabled?: boolean = false;
+  @Input() size?: 'sm' | 'md' | 'lg' = 'sm';
```
Comment (unnamjolner) [inline src/kirby/components/floating-action-button/floating-action-button.component.tns.ts]: We normally use the short versions: 's', 'm' and 'l'

File: src/kirby/components/floating-action-button/floating-action-button.component.ts (hunk 1 above)
Comment (mmmtrifork) [reply src/kirby/components/floating-action-button/floating-action-button.component.ts]: not according to src/kirby/directives/size/size.directive.ts

File: src/kirby/components/floating-action-button/floating-action-button.component.tns.ts (hunk 2 above)
Comment (mmmtrifork) [reply src/kirby/components/floating-action-button/floating-action-button.component.tns.ts]: not according to src/kirby/directives/size/size.directive.ts

File: src/kirby/components/floating-action-button/floating-action-button.component.ts (hunk 1 above)
Comment (jakobe) [reply src/kirby/components/floating-action-button/floating-action-button.component.ts]: @unnamjolner We are in the process of aligning this across components (to all use 2-letter abbrevation)

File: src/kirby/components/floating-action-button/floating-action-button.component.scss [hunk 3]
```diff
@@ -3,10 +3,25 @@
 // Ionic fab can be customized by using the css properties listed in the CSS Custom Properties section of the below docs page:
 // https://ionicframework.com/docs/api/fab-button
 
+@mixin fab-size($fab-size) {
+    width: #{$fab-size}px;
```
Comment (jakobe) [inline src/kirby/components/floating-action-button/floating-action-button.component.scss]: Pretty sure the `size()` function returns px-values...

File: src/kirby/components/floating-action-button/floating-action-button.component.scss [hunk 4]
```diff
@@ -3,10 +3,25 @@
...
+    height: #{$fab-size}px;
```
Comment (jakobe) [inline src/kirby/components/floating-action-button/floating-action-button.component.scss]: Pretty sure the `size()` function returns px-values...

File: src/kirby/components/floating-action-button/floating-action-button.component.scss [hunk 5]
```diff
@@ -3,10 +3,25 @@
...
+}
+
 ion-fab-button {
     --box-shadow: 0 10px 20px -10px #{get-color('primary-shade')};
-    width: 64px;
-    height: 64px;    
+    
+    &.sm {
+        @include fab-size(size('xl'));
```
Comment (jakobe) [inline src/kirby/components/floating-action-button/floating-action-button.component.scss]: Consider readability...
```suggestion
        width: size('xl');
        height: size('xl');
```

File: src/kirby/components/floating-action-button/floating-action-button.component.tns.scss [hunk 6]
```diff
@@ -1,25 +1,39 @@
...
+}
+
+@mixin fab-wrapper($fab-size) {
+  @include fab-size($fab-size * (3/2));
+  .fab-shadow {
+    @include fab-size($fab-size);
+    .fab {
+      @include fab-size($fab-size);
+      background-color: get-color('primary');
+      border-radius: 32;
```
Comment (jakobe) [inline src/kirby/components/floating-action-button/floating-action-button.component.tns.scss]: Should be relative to fab-size OR very large...

File: src/app/showcase/floating-action-button-showcase/floating-action-button-showcase.component.ts [hunk 7]
```diff
@@ -24,5 +24,11 @@ export class FloatingActionButtonShowcaseComponent {
       defaultValue: 'false',
       inputValues: ['boolean'],
     },
+    {
+      name: 'size',
+      description: 'The size of the button.',
+      defaultValue: 'sm',
```
[the remark this answers is not in the document]
Comment (mmmtrifork) [reply src/app/showcase/floating-action-button-showcase/floating-action-button-showcase.component.ts]: I was thinking that we can re-adjust the sizes later, once we decide on what should be small, medium and large

File: src/kirby/components/floating-action-button/floating-action-button.component.scss (hunk 3 above)
Comment (mmmtrifork) [reply src/kirby/components/floating-action-button/floating-action-button.component.scss]: unfortunately it no longer works if i don't specify it's a px-value

File: src/kirby/components/floating-action-button/floating-action-button.component.scss (hunk 4 above)
Comment (mmmtrifork) [reply src/kirby/components/floating-action-button/floating-action-button.component.scss]: unfortunately it no longer works if i don't specify it's a px-value

File: src/kirby/components/floating-action-button/floating-action-button.component.scss (hunk 5 above)
Comment (mmmtrifork) [reply src/kirby/components/floating-action-button/floating-action-button.component.scss]: I think it was readable enough (perhaps we could rename the mixin), but i'll change it to your suggestion

File: src/app/showcase/floating-action-button-showcase/floating-action-button-showcase.component.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (jakobe) [reply src/app/showcase/floating-action-button-showcase/floating-action-button-showcase.component.ts]: Ok, I was just thinking that if I look at the default FAB button "as-is" and the FAB-button used in the modal it looks like the default is `md`  and the FAB button in modal is sized to `sm`?

[the remark this answers is not in the document]
Comment (jakobe) [reply src/app/showcase/floating-action-button-showcase/floating-action-button-showcase.component.ts]: Ok, I was just thinking that if I look at the default FAB button "as-is" and the FAB-button used in the modal it looks like the default is `md`  and the FAB button in modal is sized to `sm`?

File: src/kirby/components/floating-action-button/floating-action-button.component.scss [hunk 8]
```diff
@@ -5,8 +5,21 @@
 
 ion-fab-button {
     --box-shadow: 0 10px 20px -10px #{get-color('primary-shade')};
-    width: 64px;
-    height: 64px;    
+    
+    &.sm {
+        width: #{size('xl')}px;
```
Comment (jakobe) [inline src/kirby/components/floating-action-button/floating-action-button.component.scss]: This is the `size` function - note the `addPX`?
```
@function size($key: 'm') {
  $negative: str-index($key, '-') == 1;
  @if ($negative) {
    $key: str-slice($key, 2, -1);
  }
  $value: '';
  @if ($native == false) {
    $value: addPX(_get-map-prop($sizes, $key));
  } @else {
    $value: _get-map-prop($sizes, $key);
  }
  @return if($negative, unquote('-' + $value), $value);
}
```

Comment (mmmtrifork) [reply src/kirby/components/floating-action-button/floating-action-button.component.scss]: I still haven't gotten this to work, so i kept the explicit px

File: src/app/showcase/floating-action-button-showcase/floating-action-button-showcase.component.ts [hunk 9]
```diff
@@ -24,5 +24,11 @@ export class FloatingActionButtonShowcaseComponent {
       defaultValue: 'false',
       inputValues: ['boolean'],
     },
+    {
+      name: 'isFloating',
+      description: 'Determines whether the button will be floating or not.',
+      defaultValue: 'true',
+      inputValues: ['boolean'],
```
Comment (jakobe) [inline src/app/showcase/floating-action-button-showcase/floating-action-button-showcase.component.ts]:
```suggestion
      inputValue: 'boolean',
```

File: src/kirby/components/floating-action-button/floating-action-button.component.html [hunk 10]
```diff
@@ -1,3 +1,3 @@
-<ion-fab-button [ngClass]="{ 'disabled': disabled, 'no-shadow': !showShadow }" [disabled]="disabled" tabindex="-1">
+<ion-fab-button class="{{ isFloating ? '' : 'not-floating' }}" [ngClass]="{ 'disabled': disabled, 'no-shadow': !showShadow }" [disabled]="disabled" tabindex="-1">
```
Comment (jakobe) [inline src/kirby/components/floating-action-button/floating-action-button.component.html]:
```suggestion
<ion-fab-button [ngClass]="{ 'disabled': disabled, 'no-shadow': !showShadow , 'not-floating': !isFloating}" [disabled]="disabled" tabindex="-1">
```

File: src/kirby/components/floating-action-button/floating-action-button.component.tns.scss [hunk 11]
```diff
@@ -1,25 +1,33 @@
...
-    height: 64;
-    text-align: center;
-    vertical-align: middle;
+@mixin fab-size($fab-size) {
+  width: #{$fab-size};
+  height: #{$fab-size};
+}
+
+@mixin fab-wrapper($fab-size, $color) {
+  @include fab-size($fab-size * (3/2));
```
Comment (jakobe) [inline src/kirby/components/floating-action-button/floating-action-button.component.tns.scss]: ??

File: src/kirby/components/floating-action-button/floating-action-button.component.tns.scss [hunk 12]
```diff
@@ -1,25 +1,33 @@
...
+  }
 }
 
 .fab-wrapper {
-    width:96;
-    height:96;
+  @include fab-wrapper(64, 'primary');
+  
+  &.not-floating {
+    @include fab-wrapper(size('xl'), 'light');
```
Comment (jakobe) [inline src/kirby/components/floating-action-button/floating-action-button.component.tns.scss]:
```suggestion
    @include fab-wrapper(40, 'light');
```

File: src/kirby/components/floating-action-button/floating-action-button.component.scss [hunk 13]
```diff
@@ -6,14 +6,21 @@
...
+    }
 
     &.disabled {
         --background: #{get-color('medium')};
         --box-shadow: 0;
     }
 
-    &.no-shadow {
+    &.not-floating {
+        width: #{size('xl')}px;
```
Comment (jakobe) [inline src/kirby/components/floating-action-button/floating-action-button.component.scss]:
```suggestion
        width: 40px;
```

File: src/kirby/components/floating-action-button/floating-action-button.component.scss [hunk 14]
```diff
@@ -6,14 +6,21 @@
...
+        height: #{size('xl')}px;
```
Comment (jakobe) [inline src/kirby/components/floating-action-button/floating-action-button.component.scss]:
```suggestion
        height: 40px;
```

File: src/app/showcase/floating-action-button-showcase/floating-action-button-showcase.component.ts (hunk 9 above)
Comment (mmmtrifork) [reply src/app/showcase/floating-action-button-showcase/floating-action-button-showcase.component.ts]: ShowcaseProperty class has only inputValues: string[];

**Pull request merged**